docker/build/assets.Dockerfile

```diff
@@ -94,18 +95,66 @@ ARG release_version=
 ENV CUDA_QUANTUM_VERSION=$release_version
 
 RUN cd /cuda-quantum && source scripts/configure_build.sh && \
-    ## [>CUDAQuantumBuild]
+    ## [>CUDAQuantumCppBuild]
     CUDAQ_WERROR=false \
     CUDAQ_PYTHON_SUPPORT=OFF \
     CUDAHOSTCXX="$CXX" \
     CUDAQ_ENABLE_STATIC_LINKING=true \
     LDFLAGS='-static-libgcc -static-libstdc++' \
```

Comment:
All of these are optional (and I intend to delete the static linking from here as well). However, this is concretely how binaries are built that do have minimal prerequisites (mostly just glibc) and work across a range of operating systems. The purpose of the document is not to go through everything that could be done, but to give one clear and tested way of building binaries that work in "all" cases (at least all cases that are satisfying the necessary/documented requirements).

Reply:
I'm not suggesting a list of all possible permutations. No one would read it.

My concern is the implication here. Is this a bare minimum (which is typical in build directions) or is it something else?

Many software packages do the following.
- A quick start with some absolute bare bones set of commands with only the required options.
- Followed by a list of the most common options that most users should be aware of.

Reply:
CUDAHOSTCXX is needed to successfully build in a generic environment. The reason for that is that 
1) If I set it do CXX by default in either the CMake files or the build_cudaq script, then the build will fail if gcc12 or newer is used, since CUDA does not officially support that compiler. 
2) If I don't set it in the minimal environment used for the build here, then the build will fail validation because it will pick up on older CUDA version and fail the CUDA Compiler works check.
I am open to how to better solve that, but I didn't find a good way to make it work in general except for explicitly setting it as needed - if you have a better options, I am all ears. 

Similarly, setting the flags to include the C++ standard library improves compatibility across the board. Specifically, without this, it would not be possible to build CUDA Quantum in a build environment, where C++20 is/needs to be available, and then use it on a system with C++17 only support. This way it is. I think we could reasonably get rid of this if we could also build with C++17 only support, but as discussed, we won't make a goal. 

Generally, keep in mind that the instructions are how they are because they are written to allow building on a different system than deploying/using it. This is an important requirement, because bigger HPC/cloud system will need to do this, and the targeted host system doesn't necessarily have access to the internet (which would be a huge pain for the build that downloads easily 10+ dependencies).